src/media/local-video-track.ts

```diff
@@ -0,0 +1,75 @@
+import { LocalTrack } from './local-track';
+
+export enum FacingMode {
+  user = 'user',
+  environment = 'environment',
+}
+
+export enum OptimizationMode {
+  motion = 'motion',
+  detail = 'detail',
+}
+
+export enum DisplaySurface {
+  browser = 'browser',
+  monitor = 'monitor',
+  window = 'window',
+}
+
+type BaseVideoConstraints = Pick<
+  MediaTrackConstraints,
+  'aspectRatio' | 'height' | 'width' | 'frameRate' | 'deviceId'
+>;
+
+type BaseDisplayConstraints = Pick<
+  MediaTrackConstraints,
+  'aspectRatio' | 'height' | 'width' | 'frameRate' | 'suppressLocalAudioPlayback'
+>;
+
+export interface CameraConstraints extends BaseVideoConstraints {
+  FacingMode?: FacingMode;
+  optimizationMode?: OptimizationMode;
+}
+
+export interface DisplayConstraints extends BaseDisplayConstraints {
+  displaySurface?: DisplaySurface;
+  logicalSurface?: boolean;
+}
+
+/**
+ * A class to map resolution with video constraints.
+ */
+export const StaticVideoEncoderConstraints: { [key: string]: CameraConstraints } = {
```

Comment:
I think the use of 'encoder' here feels a bit inconsistent now with the other types.  I'd just call this something like `PresetCameraConstraints`?

Reply:
done with the name change 

for the pick scenario the value add we are providing by adding enums like

```
export enum DisplaySurface {
  browser = 'browser',
  monitor = 'monitor',
  window = 'window',
}
```

is the developer know what enum to use , currently its just string which comes out of MediaTrackConstraints